lib/sparse/src/index/search_context.rs

```diff
@@ -139,25 +171,30 @@ impl<'a> SearchContext<'a> {
     /// b,  21, 34, 60, 200
     /// b,  30, 34, 60, 230
     fn advance(&mut self) -> Option<ScoredPointOffset> {
-        // Get current min record id from all posting list iterators
-        let current_min_record_id = *self.min_record_ids.peek()?;
+        // get current min tracker from all posting list iterators
+        let min_tracker = self.min_record_ids.peek()?;
+        // current record_id to score
+        let current_min_record_id = min_tracker.min_record_id;
         let mut score = 0.0;
-
-        // Iterate to advance matching posting iterators
```

Comment:
This is the important change explaining the gain.
Previously we were iterating over all posting iterator heads to find the min record id.
Gets slower as the query grows.